app/controllers/listing_endorsement_controller.rb

```diff
@@ -0,0 +1,25 @@
+class ListingEndorsementsController < ApplicationController
+  before_action :raise_suspended, only: %i[create update]
+  before_action :authenticate_user!, only: %i[create update]
+  after_action :verify_authorized, only: %i[update]
+
+  def create
+    @endorsement = ListingEndorsement.create(content: params[:content],
+                                             user_id: current_user.id,
+                                             classified_listing_id: params[:classified_listing_id])
+    if @endorsement.persisted?
+      Notification.send_new_endorsement_notifications_without_delay(@endorsement)
+      render json: { status: "created" }, status: :ok
+    else
+      render json: { error: "an error occured with the endorsement" }, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    @endorsement = ListingEndorsement.find(params[:id])
+
+    authorize @endorsement
+
+    @endorsement.update(approved: true) unless @endorsement.approved
```

Comment:
oh oh, I felt not approving == unapproving, since the approved column in the DB is false by default and we only care about column with true (approved endorsement). so unapproving really does nothing :thinking:  @rhymes 

What do you think?

Reply:
I think we should allow that a user changes their mind and removes the approval from an endorsement. So going from true to false

Reply:
oh I see, got that.